demos/unity_states/unity/Assets/Scripts/BallStateDataSourceUnity.cs

```diff
@@ -115,15 +112,22 @@ public ulong color
     {
         get
         {
-            // Convert to 16-bit raw RGBA
-            UnityEngine.Color c = this.meshRenderer.materials[0].color;
-            ushort r = (ushort)(c.r * 65535);
-            ushort g = (ushort)(c.g * 65535);
-            ushort b = (ushort)(c.b * 65535);
-            ushort a = (ushort)(c.a * 65535);
-
-            ulong rgba = (ulong)((r << 48) | (g << 32) | (b << 16) | a);
-            return rgba;
+            if (null == this.meshRenderer || this.meshRenderer.materials.Length == 0)
+            {
+                return 0;
```

Comment:
not a fan of magic values. Better to throw an exception or even better, return an Optional. Also document that this can happen in the docstring.